src/typ.rs

```diff
@@ -2365,6 +2365,16 @@ fn infer_clause_guard(
             })
         }
 
+        ClauseGuard::Tuple { location, elems , typ} => {
+
+            // TODO: How to do the type check here?
```

Comment:
For typing tuples we would want to call `infer_clause_aguard` on each element in the tuple to get the types elements. Once we have that we can take the `typ()` from each one to make a `Type::Tuple` to be the type for this tuple.